katdal/visdatav4.py

```diff
@@ -411,26 +417,15 @@ def _set_keep(self, time_keep=None, freq_keep=None, corrprod_keep=None,
             stage1 = (self._time_keep, self._freq_keep, self._corrprod_keep)
             if update_all:
                 # Cache dask graphs for the data fields
-                self._vis = DaskLazyIndexer(self.source.data.vis, stage1)
-                self._weights = DaskLazyIndexer(self.source.data.weights, stage1)
-                if self._applycal:
-                    add_applycal_transform(self._vis, self.sensor,
-                                           self.corr_products, self._applycal,
-                                           apply_vis_correction)
-                    add_applycal_transform(self._weights, self.sensor,
-                                           self.corr_products, self._applycal,
-                                           apply_weights_correction)
+                self._vis = DaskLazyIndexer(self._corrected.vis, stage1)
+                self._weights = DaskLazyIndexer(self._corrected.weights, stage1)
             flag_transforms = []
             if ~self._flags_select != 0:
                 # Copy so that the lambda isn't affected by future changes
                 select = self._flags_select.copy()
                 flag_transforms.append(lambda flags: da.bitwise_and(select, flags))
             flag_transforms.append(lambda flags: flags.view(np.bool_))
             self._flags = DaskLazyIndexer(self.source.data.flags, stage1, flag_transforms)
```

Comment:
`self._corrected.flags` instead?

Reply:
Yikes, good catch. Will fix and add a unit test.

Reply:
Hmm, unit test will be tricky since we have no unit tests that construct a dummy VisibilityDataV4. But I will fix.